File: app/controllers/home_controller.rb
class HomeController < ApplicationController
	before_action :authenticate_user!
	def index
		@all_patients = Patient.all
		@all_doctors = Doctor.all
		@all_appointments = Appointment.all
	end

	def create_patient
		patient_details = JSON.parse(params["patient_details"])
		create_new_patient = Patient.create(
			:name => patient_details["name"],
			:phone => patient_details["number"]
		)
	end

	def create_doctor
		doctor_details = JSON.parse(params["doctor_details"])
		create_new_doctor = Doctor.create(
			:name => doctor_details["name"],
			:phone => doctor_details["number"],
			:specialization => doctor_details["specialization"]
		)
	end

	def create_appointment
		appointment_details = JSON.parse(params["appointment_details"])
		create_new_appointment = Appointment.create(
			:doctor_name => appointment_details["doctor_name"],
			:patient_name => appointment_details["patient_name"],
			:disease => appointment_details["disease"]
		)
	end
	def show
		
	end
end

File: app/assets/javascripts/angular/controller/homeController.js
// management.controller('homeController',["$scope","Home", function($scope,Home){

//   $scope.initJson = function() {
//     if (typeof json != 'undefined') {
//       angular.forEach(json, function(obj_value, obj_key) {
//         $scope[obj_key] = obj_value;
//       });
//     }
//   }
//   $scope.initJson();

//   $scope.createPatient = function(data) {
//     Home.$get("/home/create_patient", {patient:data}).then(function(data) {
//       console.log("hi")
//     })
//   }


// }])



management.controller('homeController',["$scope","Home","$window", function($scope,Home,$window){
    $scope.initJson = function() {
      if (typeof json != 'undefined') {
        angular.forEach(json, function(obj_value, obj_key) {
          $scope[obj_key] = obj_value;
        });
      }
    }
    $scope.initJson();

    $scope.createPatient = function(details) {
      var patient_details = { patient_details: details }

      Home.$get("/home/create_patient", patient_details).then(function(patient) {
        $window.location.reload()
      })
    }

    $scope.createDoctor = function(details) {
      var doctor_details = { doctor_details: details }

      Home.$get("/home/create_doctor", doctor_details).then(function(doctor) {
        $window.location.reload()
      })
    }

    $scope.createAppointment = function(details) {
      var appointment_details = { appointment_details: details }

      Home.$get("/home/create_appointment", appointment_details).then(function(appointment) {
        $window.location.reload()
      })
    }

  }])
